libs/langgraph/langgraph/pregel/remote.py

```diff
@@ -0,0 +1,360 @@
+        *,
+        xray: Union[int, bool] = False,
+    ) -> DrawableGraph:
+        graph = self.sync_client.assistants.get_graph(
+            assistant_id=self.graph_id,
+            xray=xray,
+        )
+
+        # construct DrawableNodes
+        nodes = {}
+        for node in graph["nodes"]:
+            node_id = node["id"]
+            nodes[node_id] = DrawableNode(
+                id=node_id,
+                name=node.get("name", node_id),
+                data=node.get("data", {}),
+                metadata=node.get("metadata"),
+            )
+
+        return DrawableGraph(
+            nodes=nodes,
+            edges=[DrawableEdge(**edge) for edge in graph["edges"]],
+        )
+
+    async def aget_graph(
+        self,
+        config: Optional[RunnableConfig] = None,
+        *,
+        xray: Union[int, bool] = False,
+    ) -> DrawableGraph:
+        graph = await self.client.assistants.get_graph(
+            assistant_id=self.graph_id,
+            xray=xray,
+        )
+
+        # construct DrawableNodes
+        nodes = {}
+        for node in graph["nodes"]:
+            node_id = node["id"]
+            nodes[node_id] = DrawableNode(
+                id=node_id,
+                name=node.get("name", node_id),
+                data=node.get("data", {}),
+                metadata=node.get("metadata"),
+            )
+
+        return DrawableGraph(
+            nodes=nodes,
+            edges=[DrawableEdge(**edge) for edge in graph["edges"]],
+        )
+
+    def get_subgraphs(
+        self, namespace: Optional[str] = None, recurse: bool = False
+    ) -> Iterator[tuple[str, "PregelProtocol"]]:
+        subgraphs = self.sync_client.assistants.get_subgraphs(
+            assistant_id=self.graph_id,
+            namespace=namespace,
+            recurse=recurse,
+        )
+        for namespace, graph_schema in subgraphs.items():
+            remote_subgraph = self.copy({"graph_id": graph_schema["graph_id"]})
+            yield (namespace, remote_subgraph)
+
+    async def aget_subgraphs(
+        self, namespace: Optional[str] = None, recurse: bool = False
+    ) -> AsyncIterator[tuple[str, "PregelProtocol"]]:
+        subgraphs = await self.client.assistants.get_subgraphs(
+            assistant_id=self.graph_id,
+            namespace=namespace,
+            recurse=recurse,
+        )
+        for namespace, graph_schema in subgraphs.items():
+            remote_subgraph = self.copy({"graph_id": graph_schema["graph_id"]})
+            yield (namespace, remote_subgraph)
+
+    def _create_state_snapshot(self, state: ThreadState) -> StateSnapshot:
+        tasks = []
+        for task in state["tasks"]:
+            interrupts = []
+            for interrupt in task["interrupts"]:
+                interrupts.append(Interrupt(**interrupt))
+
+            tasks.append(
+                PregelTask(
+                    id=task["id"],
+                    name=task["name"],
+                    path=tuple(),
```

Comment:
why empty tuple?

Reply:
I was mapping the fields from `ThreadTask` to `PregelTask`, but there's no `path` field in `ThreadTask`.
```python
class ThreadTask(TypedDict):
    """Represents a task within a thread."""

    id: str
    name: str
    error: Optional[str]
    interrupts: list[dict]
    checkpoint: Optional[Checkpoint]
    state: Optional["ThreadState"]


class PregelTask(NamedTuple):
    id: str
    name: str
    path: tuple[Union[str, int], ...]
    error: Optional[Exception] = None
    interrupts: tuple[Interrupt, ...] = ()
    state: Union[None, RunnableConfig, "StateSnapshot"] = None
```

What's supposed to be in `path` or how do we get the `path`?

Reply:
ah, because its an internal thing I removed from the api, I guess now that comes back to bite me